src/shared/condition.c

```diff
@@ -138,6 +140,53 @@ static int condition_test_kernel_command_line(Condition *c) {
         return false;
 }
 
+static int condition_test_user(Condition *c) {
+        uid_t id;
+        int r;
+        _cleanup_free_ const char * username = NULL;
+
+        assert(c);
+        assert(c->parameter);
+        assert(c->type == CONDITION_USER);
+
+        r = parse_uid(c->parameter, &id);
+        if (r >= 0)
+                return id == getuid() || id == geteuid();
+        else if (r != -EINVAL)
+                return r;
+
+        if (streq("@system", c->parameter)) {
+                return getuid() <= SYSTEM_UID_MAX || geteuid() <= SYSTEM_UID_MAX;
+        }
+
+        username = getusername_malloc();
+        if (!username)
+                return -ENOMEM;
+
+        return streq(username, c->parameter);
+}
+static int condition_test_group(Condition *c) {
+        gid_t id;
+        int r;
+
+        assert(c);
+        assert(c->parameter);
+        assert(c->type == CONDITION_GROUP);
+
+        r = parse_gid(c->parameter, &id);
+        if (r >= 0)
+                return in_gid(id);
+        else if (r != -EINVAL)
+                return r;
+
+        // Avoid any NSS lookups if we are PID1
+        if (getpid() == 1) {
```

Comment:
I added this check here, because `c->parameter` might contain any random name, and thus `in_group` will happily ask NSS. Not sure if this is the cleaner way to do this.